src/components/DatePicker/DatePicker.tsx

```diff
@@ -0,0 +1,259 @@
+  dateFormat?: string;
+  /** Populates the `HelperErrorText` component in the "From" `TextInput` component. */
+  helperText?: string;
+  /** Populates the `HelperErrorText` component in the "To" `TextInput` component. */
+  helperTextTo?: string;
+  /** Populates the `HelperErrorText` error state for both "From" and "To" input components. */
+  errorText?: string;
+  /** Helper for modifiers array; adds 'errored' styling. */
+  errored?: boolean;
+  /** Adds the 'required' property to the input element(s). */
+  required?: boolean;
+  /** Adds the 'disabled' property to the input element(s). */
+  disabled?: boolean;
+  /** Modifiers array for use with BEM. See how to work with modifiers and BEM here: http://getbem.com/introduction/ */
+  modifiers?: string[];
+  /** BlockName for use with BEM. See how to work with blockNames and BEM here: http://getbem.com/introduction/ */
+  blockName?: string;
+  /** Value name for the input fields. The respective "From" and "To" fields appends "-from" and "-to". */
+  name?: string;
+}
+
+/**
+ * A Design System `TextInput` component that is used as the custom input
+ * element for the `DatePicker` plugin. This uses `forwardRef` because,
+ * internally, the `DatePicker` plugin needs to keep track of the input data.
+ * @note This is only used internally for this file.
+ * @note we don't declare an `onClick` prop but the `DatePicker` plugin takes
+ * care of that. That is used and needed to open/close the popup calendar.
+ * @note we can't use the prop name "required" for the `required` prop we need.
+ * Instead, we are calling it `dsRequired`. This is because the `DatePicker`
+ * plugin internally makes use of the "required" name so we need a new name.
+ */
+const CustomTextInput = forwardRef<TextInputRefType, CustomTextInputProps>(
+  (
+    {
+      showLabel,
+      onChange,
+      value,
+      labelText,
+      disabled,
+      dsRequired,
+      errored,
+      helperText,
+      errorText,
+      onClick,
+      attributes,
+    },
+    ref: React.Ref<TextInputRefType>
+  ) => (
+    <TextInput
+      showLabel={showLabel}
+      onChange={onChange}
+      value={value}
+      labelText={labelText}
+      disabled={disabled}
+      required={dsRequired}
+      errored={errored}
+      helperText={helperText}
+      errorText={errorText}
+      ref={ref}
+      attributes={{ ...attributes, onClick }}
+    />
+  )
+);
+
+/**
+ * This is the wrapper component that is rendered as the parent for the input
+ * element(s). For a single date input field, we need to render a "div", while
+ * for a date range we render a "fieldset".
+ * @note This is only used internally for this file.
+ */
+const DatePickerWrapper: React.FC<DatePickerWrapperProps> = ({
+  id,
+  dateRange,
+  showLabel,
+  labelText,
+  className = "",
+  children,
+}) =>
+  dateRange ? (
+    <fieldset id={`dateRange-${id}`} className={className}>
+      {showLabel && <legend>{labelText}</legend>}
```

Comment:
Similar to `TextInput`, when `showLabel` is set to `false`, appropriate accessibility attributes need to be added.  You might consider dropping `fieldset` altogether and using an altered `div` solutions for both.

**Visible Label**
```html
<div id={`dateRange-${id}`} className={className} role="group" aria-labelledby="the-label">
  <Label id="the-label">The Custom Label</Label>
  ...
```

**Hidden Visible**

```html
<div id={`dateRange-${id}`} className={className} role="group" aria-label="The Custom Label">
```

Reply:
I'll stick to the `fieldset` since that's in the spec for the date range. That sounds good for the accessible label. Will update.